### PR TITLE
feat: mfa

### PR DESCRIPTION
**Pull request by TorstenDittmann**
## What does this PR do?

- mfa implementation

## Test Plan

- not there yet

## Checklist

- [x] Have you read the [Contributing Guidelines on issues](https://github.com/appwrite/appwrite/blob/master/CONTRIBUTING.md)?
- [x] If the PR includes a change to an API's metadata (desc, label, params, etc.), does it also include updated API specs and example docs?

### CODE REVIEW

File: app/controllers/api/account.php [hunk 1]
```diff
@@ -3413,6 +3436,389 @@
...
+    ->inject('requestTimestamp')
+    ->inject('response')
+    ->inject('user')
+    ->inject('dbForProject')
+    ->inject('queueForEvents')
+    ->action(function (bool $mfa, ?\DateTime $requestTimestamp, Response $response, Document $user, Database $dbForProject, Event $queueForEvents) {
+
+        $user->setAttribute('mfa', $mfa);
+
+        $user = $dbForProject->withRequestTimestamp($requestTimestamp, fn () => $dbForProject->updateDocument('users', $user->getId(), $user));
```
Comment (abnegate) [inline app/controllers/api/account.php:3467]: I don't think we need `withRequestTimestamp` (also might change before we do)

File: app/controllers/api/account.php [hunk 2]
```diff
@@ -3413,6 +3436,389 @@
...
+        };
+
+        $otp->setLabel($user->getAttribute('email'));
+        $otp->setIssuer($project->getAttribute('name'));
+
+        $backups = Provider::generateBackupCodes();
+
+    if ($user->getAttribute('totp') && $user->getAttribute('totpVerification')) {
+        throw new Exception(Exception::GENERAL_UNKNOWN, 'TOTP already exists.');
+    }
```
[the remark this answers is not in the document]
Comment (TorstenDittmann) [reply app/controllers/api/account.php:3540]: thats an issue with the current formatter, its as the formatter intended - I'll introduce pint after the release to fix those 👍🏻

File: app/config/collections.php [hunk 3]
```diff
@@ -480,6 +546,87 @@
...
+            ]
+        ],
+        'indexes' => [
+            [
+                '$id' => ID::custom('_key_user'),
+                'type' => Database::INDEX_KEY,
+                'attributes' => ['userInternalId'],
+                'lengths' => [Database::LENGTH_KEY],
+                'orders' => [Database::ORDER_ASC],
+            ]
```
[the remark this answers is not in the document]
Comment (TorstenDittmann) [reply app/config/collections.php:626]: only by the internal user or the `$id` itself 👍🏻

File: app/controllers/api/account.php (hunk 1 above)
Comment (TorstenDittmann) [reply app/controllers/api/account.php:3459]: Its used in all the other similiar endpoints :)

**Pull request merged**